src/v/kafka/server/replicated_partition.cc

```diff
@@ -485,6 +485,12 @@ replicated_partition::get_leader_epoch_last_offset_unbounded(
 ss::future<error_code> replicated_partition::prefix_truncate(
   model::offset kafka_truncation_offset,
   ss::lowres_clock::time_point deadline) {
+    if (kafka_truncation_offset() == -1) {
+        kafka_truncation_offset = high_watermark();
+    }
+    if (kafka_truncation_offset() < 0) {
+        co_return error_code::offset_out_of_range;
+    }
```

Comment:
fyi: this should work w/o calling `named_type::operator()`

https://github.com/redpanda-data/redpanda/blob/fa32a58679fed5f74a8a9a0895154ae93345cef6/src/v/utils/named_type.h#L80-L88

Reply:
Ah nice. Thanks for the fyi!